.github/workflows/install-deps.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+sudo add-apt-repository 'deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14 main' -y
+# Required for g++-13, as the latest LTS at the time of this change hasn't made it available.
+sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+
+sudo apt-get update -q
+
+sudo apt-get install \
+  g++-13 \
+  gcc-multilib \
+  libc6-dev-i386 \
+  libpq-dev \
+  libldap2-dev \
+  libsasl2-dev \
+  libssl-dev \
+  clang-14 \
+  clang-tidy-14 \
+  cppcheck
+
+ldd --version
+
+sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-14 9999
+sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-14 9999
+sudo update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-14 9999
+sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-13 9999
```

Comment:
These prerequisites looks way complex to me!
And they also look quite non-standard!
Do we really need all of these?
Are there explanation behind all of these?